### DISCUSSION TITLE
blockchain size due to too many smart contracts

### DISCUSSION THREAD

**Discussion by shashibheemanapally** [smartcontractkit/full-blockchain-solidity-course-py#679] [Q&A] [answered]
Hi, I am very new to the blockchain and I have two questions.
1. If I want to add an extra node to a blockchain, should I have history all the previous transaction blocks from the genesis block on this node?
2. If yes, what happens when a lot of smart contracts are deployed on the blockchain? doesn't the size of blockchain become very huge and become hard to setup a new node?

**Comment by cromewar** [comment, marked as answer]
Hello @shashibheemanapally 
1. It depends, when you connect your machine to the blockchain as a node you also download  a big part of the data base, but is not likely for you to have all of it, and anyways is something you don't do, usually is an automated process.
2. Kind of, if there are a lot of transactions one consequence is the increment of the gas fees, and as blockchains can manage just a limited number of transactions this could end on a saturation of the system, however the file sized we are use to manage into blockchain are actually pretty small so the storage is not a real problem, also that's why is not recommended to upload heavy files like photos, because as higher the file the higher the gas fee.

I hope this info might help.
